modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -48,5 +58,54 @@ object ConstructionService {
 
       NonEmptyList.fromList(accountIdentifiers)
     }
+
+    def transactionToOperations(transaction: Transaction): EitherT[F, ConstructionError, NonEmptyList[Operation]] = {
+      val result = for {
+        transactionAmount <- refineV[Not[Equal[0L]]](transaction.amount.value.value)
+        negativeTransactionAmount <- refineV[Not[Equal[0L]]](-1L * transaction.amount.value.value)
+        negativeOperation = Operation(
+          OperationIdentifier(OperationIndex(0L)),
```

Comment:
I see that you put `OperationIndex` manually here. I think we can improve it because you have a list of operations which are not related to each other (`relatedOperations` is None). For such case you can just use index of `NonEmptyList`:

```scala
    def transactionToOperations(transaction: Transaction): Either[Throwable, NonEmptyList[Operation]] = {
      type Refinement = Not[Equal[0L]]

      val negativeTransfer = refineV[Refinement](-transaction.amount.value.value)
        .map((transaction.source, _)) // Combine negative value with source address
      val positiveTransfer = refineV[Refinement](+transaction.amount.value.value)
        .map((transaction.destination, _)) // Combine positive value with destination address

      NonEmptyList
        .of(negativeTransfer, positiveTransfer) // Create list of operations in order
        .traverseWithLongIndexM { // Include index of element and convert it to `OperationIndex`
          case (Right((address, amount)), index) =>
            Operation(
              OperationIdentifier(OperationIndex(PosLong.unsafeFrom(index))), // We are sure that index is positive so we can use unsafeFrom here
              none,
              OperationType.Transfer,
              none,
              AccountIdentifier(address, none).some,
              Amount(AmountValue(amount), DAG).some
            ).asRight[Throwable]
          case _ => Left(MalformedTransaction)
        }
    }
```

Also the solution is a little bit more DRY

Reply:
Changed.

Reply:
@cngo-github it changes nothing, so you can omit this, but @kpudlik's written that for better readability to put focus it's negative or positive operation